source/Halibut/Transport/Streams/NetworkTimeoutStream.cs

```diff
@@ -87,42 +94,24 @@ public override async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, Cancella
         }
 #endif
 
-        async Task<T> WrapWithCancellationAndTimeout<T>(
-            Func<CancellationToken, Task<T>> action,
-            int timeout,
-            bool isRead,
-            string methodName,
-            CancellationToken cancellationToken)
+        public override void Close()
         {
-            void OnCancellationAction()
-            {
-                try
-                {
-                    inner.Close();
-                }
-                catch { }
-            };
-
-            Exception CreateExceptionOnTimeout()
-            {
-                var socketException = new SocketException(10060);
-                return new IOException($"Unable to {(isRead ? "read data from" : "write data to")} the transport connection: {socketException.Message}.", socketException);
-            }
-
-            return await CancellationAndTimeoutTaskWrapper.WrapWithCancellationAndTimeout(
-                action,
-                OnCancellationAction,
-                CreateExceptionOnTimeout,
-                TimeSpan.FromMilliseconds(timeout),
-                methodName,
-                cancellationToken);
+            ThrowIfAlreadyTimedOut();
+            
+            inner.Close();
         }
 
-        public override void Close() => inner.Close();
+        public override async Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        {
+            ThrowIfAlreadyTimedOut();
+            
+            await inner.CopyToAsync(destination, bufferSize, cancellationToken);
```

Comment:
This will be a problem since the `inner` wont respect timeouts.

```
await base.CopyToAsync(destination, buffferSize, cancellationToken)
```

Would fix that.

If this is not fixed in this PR at least mark it as TODO before merging.